jetty-server/src/main/java/org/eclipse/jetty/server/ForwardedRequestCustomizer.java

```diff
@@ -598,16 +598,18 @@ public void handleSslSessionId(HttpField field)
         @SuppressWarnings("unused")
         public void handleHost(HttpField field)
         {
+            HostPort hp = new HostPort(getLeftMost(field.getValue()));
+
             if (getForwardedPortAsAuthority() && !StringUtil.isEmpty(getForwardedPortHeader()))
             {
                 if (_host == null)
-                    _host = new PossiblyPartialHostPort(getLeftMost(field.getValue()));
+                    _host = new PossiblyPartialHostPort(hp.getHost(), hp.getPort());
                 else if (_host instanceof PortSetHostPort)
-                    _host = new HostPort(HostPort.normalizeHost(getLeftMost(field.getValue())), _host.getPort());
+                    _host = new HostPort(hp.getHost(), _host.getPort());
```

Comment:
This would make it the same as `handlePort`
```suggestion
                    _host = new HostPort(hp.getHost(), hp.getPort() >0 ? hp.getPort() : _host.getPort());
```

alternately take the `port <= 0` test out of `handlePort` to get the alternate precedence.

Reply:
Done.